src/test/java/org/jenkinsci/plugins/azurekeyvaultplugin/AzureCredentialsProviderTest.java

```diff
@@ -0,0 +1,51 @@
+package org.jenkinsci.plugins.azurekeyvaultplugin;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import static junit.framework.TestCase.assertEquals;
+import static junit.framework.TestCase.fail;
+
+public class AzureCredentialsProviderTest {
+    @Rule
+    public ExpectedException expectedException = ExpectedException.none();
+
+    @Test
+    public void testGenerateKeyvaultItemName_wrong_pattern() throws Throwable {
+        expectedException.expect(AzureKeyVaultException.class);
+        expectedException.expectMessage("Wrong pattern for key vault item id.");
+        AzureCredentialsProvider provider = new AzureCredentialsProvider();
+        try {
+            Method generateKeyvaultItemName = AzureCredentialsProvider.class.getDeclaredMethod("generateKeyvaultItemName"
```

Comment:
Could you just make it package-private which will make it visible for testing but not expose it externally